src/mixtures/mixturemodel.jl

```diff
@@ -48,7 +48,7 @@ component_type(d::AbstractMixtureModel{VF,VS,C}) where {VF,VS,C} = C
 
 Get a list of components of the mixture model `d`.
 """
-components(d::AbstractMixtureModel)
+components(d::AbstractMixtureModel) = [component(d, k) for k in 1:ncomponents(d)]
```

Comment:
What is this actually for? `component` is - I think - not defined for the `AbstractMixtureModel` and `MixtureModel` already has a method `components`

Reply:
From the documentation of `AbstractMixture`:
```julia
  All subtypes of `AbstractMixtureModel` should implement the following methods:
  - ncomponents(d): the number of components
  - component(d, k):  return the k-th component
  - probs(d):       return a vector of prior probabilities over components.
```

Given these basic functions, a fallback for `components` can be defined at the level of AbstractMixture